context/app/static/js/components/detailPage/visualization/VisualizationTracker/utils.ts

```diff
@@ -0,0 +1,159 @@
+import hasKey from 'js/helpers/hasKey';
+import { MutableRefObject } from 'react';
+import { trackEvent } from 'js/helpers/trackers';
+import { VitessceInteraction } from './types';
+
+/**
+ * Titles of components in the Vitessce UI
+ * @see https://vitessce.io/docs/components/
+ */
+const visualizationComponents = [
+  'Spatial Layers', // layerController
+  'Spatial', // spatial
+  'Scatterplot', // scatterplot
+  'Description', // description
+  'Data Set', // description alternative
+  'Heatmap', // heatmap
+  'Expression by Cell Set', // obsSetFeatureValueDistribution
+  'Cell Sets', // obsSets
+  'Gating', // gating
+  'Genomic Profiles', // genomicProfiles
+  'Expression Levels', // featureList
+  'Antigen List', // featureList alternative
+  'Cell Set Sizes', // obsSetSizes
+  'Status', // status
+  'Expression Histogram', // featureHistogram
+];
+
+/**
+ * Retrieves the most relevant human-readable identifier for a given element.
+ * @param target The target element.
+ * @returns The title or component name that most accurately describes the current target.
+ */
+export function getNearestIdentifier(target: HTMLElement | null): string | null {
+  if (!target) return null;
+  // If the target has a title, return it (e.g. button controls)
+  if (target.title) {
+    // Remove excessive whitespace from the title
+    return target.title.replace(/\s+/g, ' ');
+  }
+  for (const component of visualizationComponents) {
+    if ('innerText' in target && target.innerText.includes(component)) return component;
+  }
```

Comment:
Could we just use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find and avoid the eslint rule changes?

Reply:
Done, good catch!